sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -684,31 +694,35 @@ export class MessageSession extends LinkEntity {
    */
   async close(isClosedDueToExpiry?: boolean): Promise<void> {
     try {
-      log.messageSession(
-        "[%s] Closing the MessageSession '%s' for queue '%s'.",
-        this._context.namespace.connectionId,
-        this.sessionId,
-        this.name
-      );
+      await defaultLock.acquire(this._connectionLock, async () => {
+        this._wasCloseCalled = true;
```

Comment:
How is this different from `wasCloseInitiated`?

Reply:
Oh, no good reason. I'll rename.